teachings.json

```diff
@@ -0,0 +1,409 @@
+[
+	{
+		"year": 1,
```

Comment:
Così per aggiungere un corso a scelta a più insegnamenti non siamo obbligati a duplicarne l'oggetto cambiandone solo un campo? Mi sembra questo modello generi codice ridondante. Potremmo invece dire che ciascun insegnamento ha un vettore di coppie CdL/anno?

Inoltre mi pare sia molto controintuitivo dividere per anno così anche corsi di triennali/magistrali diverse.

Reply:
Con questa struttura pensavo più ad avere un elenco con tutti gli insegnamenti esistenti, e poi in degrees richiamare ed associare eventuali teachings di altri insegnamenti (appunto per non avere doppioni)

Reply:
> Con questa struttura pensavo più ad avere un elenco con tutti gli insegnamenti esistenti, e poi in degrees richiamare ed associare eventuali teachings di altri insegnamenti (appunto per non avere doppioni)

Quindi in degrees poi smetteremo di usare ref e li prenderemo a mano giusto (siccome non si può usare un semplice filtro dal momento che un insegnamento può essere in più CdL)? A quel punto sposterei comunque gli anni in degree perché l'anno non è una caratteristica univoca dell'insegnamento (magari vogliamo fare come opzionali al terzo anno esami interessanti di ingegneria del secondo), ma una sottostruttura del corso di laurea. E rimouverei l'id dall'insegnamento perché coincide con il primo CdL che vi fa riferimento con `optional: false`. E cioè:

`degree.json`: degrees/years/nomi-delle-repository oppure (identificano univocamente un insegnamento come dicevi tu)+optional

`teachings.json`: nomi-delle-repository + nome + chat + id-pagina-web + email

Reply:
> > Con questa struttura pensavo più ad avere un elenco con tutti gli insegnamenti esistenti, e poi in degrees richiamare ed associare eventuali teachings di altri insegnamenti (appunto per non avere doppioni)
> 
> Quindi in degrees poi smetteremo di usare ref e li prenderemo a mano giusto (siccome non si può usare un semplice filtro dal momento che un insegnamento può essere in più CdL)? 

Se assegnamo un id univoco al teaching da includere in un cdl che non è il suo corso naturale si può usare `$ref`

> A quel punto sposterei comunque gli anni in degree perché l'anno non è una caratteristica univoca dell'insegnamento (magari vogliamo fare come opzionali al terzo anno esami interessanti di ingegneria del secondo), ma una sottostruttura del corso di laurea. E rimouverei l'id dall'insegnamento perché coincide con il primo CdL che vi fa riferimento con `optional: false`. E cioè:
> 
> `degree.json`: degrees/years/nomi-delle-repository oppure (identificano univocamente un insegnamento come dicevi tu)+optional
> 
> `teachings.json`: nomi-delle-repository + nome + chat + id-pagina-web + email

Facendo così non perdiamo il senso di fare i 2 file? 

Credevo volessimo evitare di dover editare 2 file per inserire 1 insegnamento...

Reply:
> Credevo volessimo evitare di dover editare 2 file per inserire 1 insegnamento...

@VaiTon aveva detto che a lui va bene: https://github.com/csunibo/dynamik/issues/122#issuecomment-1772855776 Io lo trovo purtroppo inevitabile se vogliamo rendere n-n la relazione CdL-insegnamento + supportare lab.

> Facendo così non perdiamo il senso di fare i 2 file?

la frase sopra penso sia il vero senso di avere due file, e quello rimane intatto.

Reply:
Oks, facciamo così allora. Pomeriggio vedo di riadattare.

Per il momento lascerei `courses.json` per evitare problemi e pian piano convertire tutto. O lo sostituiamo?

Reply:
Siccome non abbiamo un sistema di versionamento fra progetti basato su release per me ci può stare. Io intanto potrei approfittarne per lavorare su `id` e `mail` e sul lato bot o qualcosa di simile. Buona idea!

Reply:
Possiamo anche sostituire, e non portare avanti i sottomoduli nei progetti che usano ancora il formato vecchio

Reply:
Mi va bene anche questo. Mi raccomando facciamo attenzione allora 😆